src/utils/parse-smart-query.ts

```diff
@@ -112,66 +131,67 @@ export const parseSmartQuery = (
                     pureValue = GeneratePureValue(value)
                     if (pureValue !== null) {
                         andQuery.push({ [key]: { $gt: pureValue } })
                     }
                     break
-                case term.includes('<'):
-                    ;[key, value] = term.split('<').map((x) => x.trim())
+                case termUpToQuote.includes('<'):
+                    ;[key, value] = SplitFirst(term, '<').map((x) => x.trim())
                     pureValue = GeneratePureValue(value)
                     if (pureValue !== null) {
                         andQuery.push({ [key]: { $lt: pureValue } })
                     }
                     break
-                case term.includes('!='):
-                    ;[key, value] = term.split('!=').map((x) => x.trim())
+                case termUpToQuote.includes('!='):
+                    ;[key, value] = SplitFirst(term, '!=').map((x) => x.trim())
                     pureValue = GeneratePureValue(value)
                     if (pureValue !== null) {
                         andQuery.push({ [key]: { $ne: pureValue } })
                     }
                     break
-                case term.includes('='):
-                    ;[key, value] = term.split('=').map((x) => x.trim())
+                case termUpToQuote.includes('='):
+                    ;[key, value] = SplitFirst(term, '=').map((x) => x.trim())
                     pureValue = GeneratePureValue(value)
                     if (pureValue !== null) {
                         andQuery.push({ [key]: pureValue })
                     }
                     break
-                case term.includes('EXISTS'):
+                case termUpToQuote.includes('EXISTS'):
                     key = term.split('EXISTS')[0].trim()
                     andQuery.push({ [key]: { $exists: true } })
                     break
-                case term.includes('DOESNT-EXIST'):
+                case termUpToQuote.includes('DOESNT-EXIST'):
                     key = term.split('DOESNT-EXIST')[0].trim()
                     andQuery.push({ [key]: { $exists: false } })
                     break
-                case term.includes('IN'):
-                    ;[key, value] = term.split('IN').map((x) => x.trim())
+                case termUpToQuote.includes('IN'):
+                    ;[key, value] = SplitFirst(term, 'IN').map((x) => x.trim())
                     if (key.includes('NOT-')) {
-                        ;[key, value] = term
-                            .split('NOT-IN')
-                            .map((x) => x.trim())
-                        queryValue = term
-                            .split('NOT-IN')
-                            .map((x) => x.trim())[1]
+                        ;[key, value] = SplitFirst(term, 'NOT-IN').map((x) =>
+                            x.trim()
+                        )
+                        // TODO make , inside quotes work
```

Comment:
What is this TODO for ? Please create a ticket for it if needed

Reply:
sure, https://github.com/dataloop-ai/components/issues/943